config.js

```diff
@@ -534,128 +534,128 @@ exports.updateFromArgv = function() {
+            } else if (e === "custom") {
                 exports.allow_custom_sizes_unranked = true;
-                for (let boardsizewidthunranked of argv.boardsizewidthunranked.split(',')) {
-                    exports.allowed_custom_boardsizewidth_unranked[boardsizewidthunranked] = true;
+                for (let e of argv.boardsizewidthunranked.split(',')) {
+                    exports.allowed_custom_boardsizewidth_unranked[e] = true;
                 }
-                for (let boardsizeheightunranked of argv.boardsizeheightunranked.split(',')) {
-                    exports.allowed_custom_boardsizeheight_unranked[boardsizeheightunranked] = true;
+                for (let e of argv.boardsizeheightunranked.split(',')) {
+                    exports.allowed_custom_boardsizeheight_unranked[e] = true;
                 }
             } else {
-                exports.allowed_sizes_unranked[boardsizeunranked] = true;
+                exports.allowed_sizes_unranked[e] = true;
             }
         }
     }
 
     if (argv.komi) {
-        for (let komi of argv.komi.split(',')) {
-            if (komi == "all") {
+        for (let e of argv.komi.split(',')) {
+            if (e === "all") {
                 exports.allow_all_komi = true;
-            } else if (komi == "automatic") {
+            } else if (e === "automatic") {
                 exports.allowed_komi[null] = true;
             } else {
-                exports.allowed_komi[komi] = true;
+                exports.allowed_komi[e] = true;
             }
         }
     }
 
     if (argv.komiranked) {
-        for (let komiranked of argv.komiranked.split(',')) {
-            if (komiranked == "all") {
+        for (let e of argv.komiranked.split(',')) {
+            if (e === "all") {
                 exports.allow_all_komi_ranked = true;
-            } else if (komiranked == "automatic") {
+            } else if (e === "automatic") {
                 exports.allowed_komi_ranked[null] = true;
             } else {
-                exports.allowed_komi_ranked[komiranked] = true;
+                exports.allowed_komi_ranked[e] = true;
             }
         }
     }
 
     if (argv.komiunranked) {
-        for (let komiunranked of argv.komiunranked.split(',')) {
-            if (komiunranked == "all") {
+        for (let e of argv.komiunranked.split(',')) {
+            if (e === "all") {
                 exports.allow_all_komi_unranked = true;
-            } else if (komiunranked == "automatic") {
+            } else if (e === "automatic") {
                 exports.allowed_komi_unranked[null] = true;
             } else {
-                exports.allowed_komi_unranked[komiunranked] = true;
+                exports.allowed_komi_unranked[e] = true;
             }
         }
     }
 
     if (argv.speed) {
-        for (let i of argv.speed.split(',')) {
-            exports.allowed_speeds[i] = true;
+        for (let e of argv.speed.split(',')) {
```

Comment:
no actually i like your idea

because speed and co are arrays, it is also clearer for the user to know that many values can be used, not just one

i dont mind changing all existing arguments with plurals, and adding depreciation message for this too :+1: @Dorus

Reply:
as for the naming, i think we should stick with one setting and apply it for all cases,

- either long names for all
- or `e` for all

but i find `e` faster and less hassle to code (no need to add general/ranked/unranked different names, and also it's universal (no need to add komi or boardsize or else, it is always simply `e`)

Reply:
but before making plurals in argv we first need to settle this PR first

so feel free to add suggestions or comments :+1:

Reply:
The variable names you use in code are independent from those used in the config/command line arguments.

I do like the idea to make it plural in the command line, but that's a completely separate issue.

There is zero reason to use `e` for all. Generally code standard is the longer the scope of a variable, the clearer the name should be. Also clearer names never hurt. It's not directly urgent to change all `e` to something long, however in places where the scope is longer than one or a few lines, and also in places where we already use long names, there is zero reason to go back to `e`.

Reply:
ok, then we should just remove all `e` and replace them with names everywhere :+1: 

i'll rebase all this then and redo it again

Reply:
But this checking is about fixing `==` vs `===`. It would really be best if you did just that, and did all the renaming stuff in a new PR. You can just put the new PR on top of this one, as this one can be pulled in 5 seconds without the renaming stuff.